src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/jpa/service/InterventionService.kt

```diff
@@ -0,0 +1,24 @@
+package uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.service
+
+import org.springframework.data.repository.findByIdOrNull
+import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.dto.InterventionDTO
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.entity.Intervention
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.repository.InterventionRepository
+import java.util.UUID
+
+@Service
+class InterventionService(val repository: InterventionRepository) {
+
+  fun createIntervention(interventionDto: InterventionDTO): Intervention {
+    return repository.save(Intervention(name = interventionDto.name))
+  }
+
+  fun getIntervention(id: UUID): Intervention? {
+    return repository.findByIdOrNull(id)
+  }
+
+  fun createInterventions(interventions: List<Intervention>): List<Intervention> {
```

Comment:
i'm slightly confused reading this between the method signatures of `createInterventions` vs `createIntervention`. why do they take different intervention objects? can we just slim this down to a single endpoint?

Reply:
It wasdone to map thru to the save and saveAll repository calls.  Happy to switch to just one.